tests/test_connection.py

```diff
@@ -604,7 +618,16 @@ def test_read_preference_from_parse(self):
     def test_multiple_connection_settings(self):
         connect("mongoenginetest", alias="t1", host="localhost")
 
-        connect("mongoenginetest2", alias="t2", host="127.0.0.1")
+        if PYMONGO_VERSION >= (4,):
+            """
+            pymongo>=4 changed how it handles connection addresses wrt replicaSets, which
+            is now how mongod is run for the test suites for transactions.
+            """
+            connect(
+                "mongoenginetest2", alias="t2", host="127.0.0.1", directConnection=True
```

Comment:
Is this change still needed? It seems unrelated to the run_in_transaction feature.

Reply:
For this test I needed to make a change in order for it to pass. It seemed to me that we needed to account for how pymongo deals with connections to replica sets (which is needed for transactions) as of >= 4.0 https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#mongoclient

Reply:
Oh thanks for the explanation. The issue is that without directConnection=True, the client will discover the real server address of the replica set member so `mongo_connections["t2"].address[0]` is equal to `localhost` not `127.0.0.1`. This seems fine, although the test could be rewritten to just do this instead of checking the address property:
```python
assert mongo_connections["t1"] is not mongo_connections["t2"]
```

Reply:
:100: great call. I went ahead and made this update as well